src/uri/mod.rs

```diff
@@ -981,6 +982,32 @@ impl<'a> PartialEq<Uri> for &'a str {
 
 impl Eq for Uri {}
 
+/// Order by `(Authority, Scheme, PathAndQuery)`, delegating to the custom
+/// case-insensitive `Ord` for [`Authority`] and [`Scheme`], and `str`
+/// byte-wise `Ord` for [`PathAndQuery`].
+impl PartialOrd for Uri {
+    #[inline]
+    fn partial_cmp(&self, other: &Self) -> Option<cmp::Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+/// Order by `(Authority, Scheme, PathAndQuery)`, delegating to the custom
+/// case-insensitive `Ord` for [`Authority`] and [`Scheme`], and `str`
+/// byte-wise `Ord` for [`PathAndQuery`].
+impl Ord for Uri {
+    fn cmp(&self, r: &Self) -> cmp::Ordering {
+        let mut ord = self.authority().cmp(&r.authority());
+        if ord == cmp::Ordering::Equal {
```

Comment:
`Ordering` has a method `then` that simplifies this pattern a lot, I think you should use it.

Reply:
Thanks, I didn't know about that. Changed in 40b780f, which includes a bench, just to confirm the `then_with` version will inline to roughly the same thing.  Perf-wise it seems to!